packages/cardano-services/src/StakePoolSearch/DbSyncStakePoolSearchProvider/queries.ts

```diff
@@ -10,47 +10,294 @@ export const findLastEpoch = `
+        ($2::numeric)
+      )::numeric
+  END AS saturation,
+  CASE
+    WHEN (COALESCE(a_stake.active_stake,0)+COALESCE(l_stake.live_stake,0))::numeric = 0::numeric
+    THEN 0::numeric
+    ELSE
+    (COALESCE(a_stake.active_stake,0)/(COALESCE(a_stake.active_stake,0)+COALESCE(l_stake.live_stake,0))) 
+  END AS active_stake_percentage,
+  CASE
+  WHEN (COALESCE(a_stake.active_stake,0)+COALESCE(l_stake.live_stake,0))::numeric = 0::numeric
+  THEN 0::numeric
+  ELSE
+  (COALESCE(l_stake.live_stake,0)/(COALESCE(a_stake.active_stake,0)+COALESCE(l_stake.live_stake,0))) 
+  END AS live_stake_percentage,
+  ph.id AS pool_hash_id 
+FROM pool_hash ph
+LEFT JOIN blocks_created bc on 
+  bc.pool_hash_id = ph.id
+LEFT JOIN delegators d on 
+  d.pool_hash_id = ph.id
+LEFT JOIN active_stake a_stake on 
+  a_stake.pool_hash_id = ph.id
+LEFT JOIN live_stake l_stake on 
+  l_stake.pool_hash_id = ph.id
+LEFT JOIN total_rewards_of_reward_acc AS tr ON
+	tr.pool_hash_id = ph.id
+LEFT JOIN total_withdraws_of_reward_acc AS tw ON
+	tw.pool_hash_id = ph.id
+LEFT JOIN owners_total_utxos otu on 
+	otu.pool_hash_id = ph.id
+where id = ANY($1)
+`;
+
 export const findPoolEpochRewards = (limit?: number) => `
 WITH epochs AS (
 	SELECT 
 		"no",
-		(extract(epoch from (end_time - start_time)) * 1000) as epoch_length
+		(extract(epoch FROM (end_time - start_time)) * 1000) AS epoch_length
 	FROM epoch 
 	ORDER BY id DESC
   ${limit !== undefined ? `LIMIT ${limit}` : ''}
 ),
-total_stake_per_epoch as (
+total_stake_per_epoch AS (
   SELECT 
-      e."no" as epoch_no,
-      sum(es.amount) as active_stake
-    from epochs e
-    left join epoch_stake es on
+      e."no" AS epoch_no,
+      sum(es.amount) AS active_stake
+    FROM epochs e
+    LEFT JOIN epoch_stake es on
       es.epoch_no  = e."no" and 
       es.pool_id = $1
-    group by e."no"
+    GROUP BY e."no"
   ),
-  total_rewards_per_epoch as (
-  select 
-    sum(r.amount) as total_amount,
-    e."no" as epoch_no
-  from  epochs e
-  left join reward r on
+  total_rewards_per_epoch AS (
+  SELECT 
+    sum(r.amount) AS total_amount,
+    e."no" AS epoch_no
+  FROM  epochs e
+  LEFT JOIN reward r on
     r.earned_epoch = e."no" and 
     r.pool_id = $1
-  group by e."no"
+  GROUP BY e."no"
   )
-  select 
+  SELECT 
     r_e.epoch_no,
     e.epoch_length,
-    COALESCE(r_e.total_amount,0) as total_rewards,
-    COALESCE (r_e.total_amount * pu.margin + pu.fixed_cost,0) as operator_fees,
-    COALESCE(total_stake.active_stake,0) as active_stake,
-    COALESCE((r_e.total_amount - ((r_e.total_amount * pu.margin) + pu.fixed_cost))
-      /total_stake.active_stake, 0) AS member_roi
-  from total_rewards_per_epoch r_e
-  join epochs e on 
+    COALESCE(r_e.total_amount,0) AS total_rewards,
+    CASE
+    WHEN pu.fixed_cost >= r_e.total_amount
+    THEN r_e.total_amount
+    ELSE (
+      COALESCE (FLOOR((r_e.total_amount-pu.fixed_cost) * pu.margin + pu.fixed_cost),0)
```

Comment:
@rhyslbw @mkazlauskas Do you think we can use this formula or has to strictly be  the `5.5.4.1` of the ledger? What differs from that is a sum factor on the margin value which is not straightforward to calculate.

Reply:
Cont https://github.com/input-output-hk/cardano-js-sdk/pull/231#discussion_r863495064